code/lib/cli/src/upgrade.ts

```diff
@@ -148,6 +148,7 @@ export const doUpgrade = async ({
   ]);
 
   const isOutdated = lt(currentVersion, latestVersion);
+  const isExactLatest = currentVersion === latestVersion;
```

Comment:
I am curious. If `isExactLatest` is `true`, isn't `isOutdated` automatically `false` and the other way around. So if `isExactLatest` is `false`, `isOutdated` is true. Therefore, isn't `isOutdated` redundant?

Reply:
I didn't want to mess with the `isOutdated` flag, considering that's was causing the bug to begin with.

**Considering this**:

user updates from `1.0.0` to `2.0.0`, whilst `3.0.0` is `latest`.
```
isOutdated=true
isLatest=false
```